File: week6/at.py
import matplotlib.pyplot as plt
import numpy as np
x=[]
y=[]
lx=int(input("Enter length of x:"))
ly=int(input("Enter length of y:"))
print("Enter elements into x:")
for i in range(lx):
	a=int(input("Enter:"))
	x.append(a)
print("Enter elements into y:")
for i in range(ly):
	a=int(input("Enter:"))
	y.append(a)
z=[]
for k in range(ly-1):
	s=0
	for n in range(lx):
		if n+k<lx:
			s=s+x[n]*y[n+k]
	z.append(s)
plt.plot(z)
plt.xlabel('Lag')
plt.ylabel('cross-correlation')
plt.title('cross-correlation between x and y')
plt.show()

File: week6/at.m
x=[]
y=[]
lx=input('Enter length of x:')
ly=input('Enter length of y:')
disp('Enter numbers into x:')
for i=1:lx
  a=input('Enter:')
  x=[x a]
endfor
disp('Enter elements into y:')
for i=1:ly
  a=input('Enter:')
  y=[y a]
endfor
z=[]
for k=0:ly-2
  s=0
  for n=1:lx
    if n+k<=lx
      s=s+x(n)*y(n+k)
    endif
  endfor
  z=[z s]
endfor
plot(z)
xlabel('Lag')
ylabel('cross-correlation')
title('cross-correlation between x and y')
